File: Stack/LinkedListStack/LinkedList.java
package DSA.Stack.LinkedListStack;

import java.util.Scanner;

class Node {
    public int data;
    public Node next;

    public Node(int value) {
        data = value;
        next = null;
    }
}

class LinkedList1 {
    private Node head;

    public LinkedList1() {
        head = null;
    }

    public void push(int value) {
        Node newNode = new Node(value);
        newNode.next = head;
        head = newNode;
    }

    public int pop() {
        if (head == null) {
            System.out.println("Stack is Empty");
            return -1;
        }
        int poppedValue = head.data;
        Node temp = head;
        head = head.next;
        temp.next = null;
        return poppedValue;
    }

    public int peek() {
        if (head == null) {
            System.out.println("Stack is Empty");
            return -1;
        }
        System.out.println("Top Element is >> " + head.data);
        return head.data;
    }

    public int count() {
        int count = 0;
        Node current = head;
        while (current != null) {
            count++;
            current = current.next;
        }
        System.out.println("Number of Stack Elements is >> " + count);
        return count;
    }

    public void destroy() {
        while (head != null) {
            pop();
        }
        System.out.println("Stack Destroyed");
    }

    public void display() {
        if (isEmpty()) {
            return;
        }
        Node current = head;
        while (current != null) {
            System.out.print(current.data + "] ");
            current = current.next;
        }
        System.out.println();
    }

    public boolean isEmpty() {
        if (head == null) {
            System.out.println("Stack is Empty");
        } else {
            System.out.println("Stack is not Empty");
        }
        return head == null;
    }
}

public class LinkedList {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int ch = 0, data = 0;
        LinkedList1 x = new LinkedList1();
        do {
            System.out.println("\n\n1.Push\n2.Pop\n3.Top\n4.isEmpty\n6.Count\n7.Destroy\n8.Display\n9.Exit \n>> ");
            ch = scanner.nextInt();
            switch (ch) {
                case 1:
                    System.out.println("Enter data >> ");
                    data = scanner.nextInt();
                    x.push(data);
                    break;
                case 2:
                    x.pop();
                    break;
                case 3:
                    x.peek();
                    break;
                case 4:
                    x.isEmpty();
                    break;
                case 6:
                    x.count();
                    break;
                case 7:
                    x.destroy();
                    break;
                case 8:
                    x.display();
                    break;
            }
        } while (ch != 9);
        scanner.close();
    }
}

File: Stack/LinkedListStack/LinkedList.py
class Node:
    def __init__(self, value):
        self.data = value
        self.next = None

class LinkedList:
    def __init__(self):
        self.head = None

    def push(self, value):
        new_node = Node(value)
        new_node.next = self.head
        self.head = new_node

    def pop(self):
        if self.head is None:
            print("Stack is Empty")
            return -1
        popped_value = self.head.data
        temp = self.head
        self.head = self.head.next
        temp.next = None
        return popped_value

    def peek(self):
        if self.head is None:
            print("Stack is Empty")
            return -1
        print("Top Element is >>", self.head.data)
        return self.head.data

    def count(self):
        count = 0
        current = self.head
        while current is not None:
            count += 1
            current = current.next
        print("Number of Stack Elements is >>", count)
        return count

    def destroy(self):
        while self.head is not None:
            self.pop()
        print("Stack Destroyed")

    def display(self):
        if self.is_empty():
            return
        current = self.head
        while current is not None:
            print(current.data, end="] ")
            current = current.next
        print()

    def is_empty(self):
        if self.head is None:
            print("Stack is Empty")
        else:
            print("Stack is not Empty")
        return self.head is None

def main():
    ch = 0
    x = LinkedList()
    while ch != 9:
        print("\n\n1.Push\n2.Pop\n3.Top\n4.isEmpty\n6.Count\n7.Destroy\n8.Display\n9.Exit \n>> ")
        ch = int(input())
        if ch == 1:
            data = int(input("Enter data >> "))
            x.push(data)
        elif ch == 2:
            x.pop()
        elif ch == 3:
            x.peek()
        elif ch == 4:
            x.is_empty()
        elif ch == 6:
            x.count()
        elif ch == 7:
            x.destroy()
        elif ch == 8:
            x.display()

if __name__ == "__main__":
    main()
